core/src/main/java/org/elasticsearch/common/inject/TypeLiteral.java

```diff
@@ -311,16 +315,19 @@ Type resolveType(Type toResolve) {
     public List<TypeLiteral<?>> getExceptionTypes(Member methodOrConstructor) {
         Type[] genericExceptionTypes;
 
-        if (methodOrConstructor instanceof Method) {
+        if (!(methodOrConstructor instanceof Method)) {
```

Comment:
@nik9000 It's not. Thank you for catching. I fixed it in c2f3019c0d3264aa15617c20965de1f81cf16e2b.